autobahn/wamp/protocol.py

```diff
@@ -457,7 +457,7 @@ def onUserError(self, fail, msg):
         """
         Implements :func:`autobahn.wamp.interfaces.ISession.onUserError`
         """
-        if isinstance(fail.value, exception.ApplicationError):
+        if hasattr(fail, 'value') and isinstance(fail.value, exception.ApplicationError):
             self.log.warn('{klass}.onUserError(): "{msg}"',
```

Comment:
When does `fail` ever not have a `.value` attribute?

Reply:
My IDE's code analyzer had detected that. That issue happens here https://github.com/crossbario/autobahn-python/blob/8cba0623b6891d253a75ea16e95d49374e0010ec/autobahn/wamp/protocol.py#L825

Reply:
Ah, good catch! That should be fixed too then .. I guess maybe a separate PR? But yeah I think `onUserError` should always be receiving `Failure`s

Reply:
^ Reported that as https://github.com/crossbario/autobahn-python/issues/1318